drivers/hmis/spec/requests/hmis/update_external_form_submission_spec.rb

```diff
@@ -0,0 +1,300 @@
+      },
+    }
+  end
+
+  context 'when submission was already reviewed' do
+    let!(:submission) { create(:hmis_external_form_submission, definition: definition, status: 'reviewed') }
+
+    it 'cannot update back to new' do
+      response, result = post_graphql({ **input, input: { status: 'new' } }) { mutation }
+      expect(response.status).to eq(200), result.inspect
+      expect(result.dig('data', 'updateExternalFormSubmission', 'errors', 0, 'fullMessage')).to eq('Cannot change status from Reviewed to New')
+    end
+  end
+
+  context 'when reviewing a submission' do
+    it 'should create CDE' do
+      expect do
+        response, result = post_graphql(input) { mutation }
+        expect(response.status).to eq(200), result.inspect
+        expect(result.dig('data', 'updateExternalFormSubmission', 'externalFormSubmission', 'status')).to eq('reviewed')
+      end.to change(Hmis::Hud::CustomDataElement, :count).by(1).
+        and not_change(Hmis::Hud::Enrollment, :count).
+        and not_change(Hmis::Hud::Client, :count)
+    end
+
+    context 'when there is an unexpected key on the input' do
+      let!(:submission) do
+        data = {
+          'foo': 'bar',
+        }.stringify_keys
+        create(:hmis_external_form_submission, raw_data: data, definition: definition)
+      end
+
+      it 'should raise an error' do
+        expect do
+          expect_gql_error(post_graphql(input) { mutation })
+        end.to not_change(Hmis::Hud::CustomDataElement, :count)
+      end
+    end
+
+    context 'when the form definition accepts client/enrollment information' do
+      let!(:definition) do
+        items = [
+          {
+            'type': 'STRING',
+            'link_id': 'first_name',
+            'mapping': {
+              'field_name': 'firstName',
+              'record_type': 'CLIENT',
+            },
+            'text': 'First name',
+          },
+          {
+            'type': 'CHOICE',
+            'pick_list_reference': 'RelationshipToHoH',
+            'link_id': 'relationship_to_hoh',
+            'mapping': {
+              'field_name': 'relationshipToHoH',
+              'record_type': 'ENROLLMENT',
+            },
+            'text': 'Relationship to HoH',
+          },
+          {
+            'type': 'STRING',
+            'link_id': 'household_id',
+            'mapping': {
+              'field_name': 'householdId',
+              'record_type': 'ENROLLMENT',
+            },
+          },
+        ]
+        create(:hmis_external_form_definition, append_items: items)
+      end
+
+      let!(:submission) do
+        data = {
+          'Client.firstName': 'Oranges',
+          'captcha_score': '1.', # also test that extraneous fields get filtered out
+          'form_definition_id': definition.id,
+          'form_content_digest': 'something random',
+        }.stringify_keys
+        create(:hmis_external_form_submission, raw_data: data, definition: definition)
+      end
```

Comment:
I think this submission should be inside the `when the submission specifies client info only` context?